.github/workflows/gcc.yml

```diff
@@ -8,85 +9,79 @@ defaults:
     shell: bash -leo pipefail {0}
 
 env:
-  cache_key: gcc2  # The number (#) following the cache_key "gcc" is to flush Action cache.
+  cache_key: gcc
   CC: gcc-10
   FC: gfortran-10
   CXX: g++-10
 
-# A note on flushing Action cache and relevance to "cache_key" above.
-# There is no way to flush the Action cache, and hence a number (#) is appended
-# to the "cache_key" (gcc).
-# If the dependencies change, increment this number and a new cache will be
-# generated by the dependency build step "setup"
-# There is a Github issue to force clear the cache.
-# See discussion on:
-# https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions
-
 # The jobs are split into:
 # 1. a dependency build step (setup), and
-# 2. a GSI-monitor build step (build)
+# 2. a GSI-monitor build step (gsi-monitor)
 # The setup is run once and the environment is cached,
-# so each build of GSI-monitor can reuse the cached dependencies to save time (and compute).
+# so each subsequent build of GSI-monitor can reuse the cached dependencies to save time (and compute).
 
 jobs:
   setup:
     runs-on: ubuntu-latest
 
     steps:
+      - name: checkout  # this is to get the ci/spack.yaml file
+        uses: actions/checkout@v3
+        with:
+            path: gsi-monitor
+
       # Cache spack, compiler and dependencies
       - name: cache-env
         id: cache-env
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: |
             spack
             ~/.spack
-          key: spack-${{ runner.os }}-${{ env.cache_key }}
-
-      - name: checkout-gsi-monitor  # This is for getting spack.yaml
-        if: steps.cache-env.outputs.cache-hit != 'true'
-        uses: actions/checkout@v2
-        with:
-            path: GSI-Monitor
+          key: spack-${{ runner.os }}-${{ env.cache_key }}-${{ hashFiles('gsi-monitor/ci/spack.yaml') }}
 
       # Install dependencies using Spack
       - name: install-dependencies-with-spack
         if: steps.cache-env.outputs.cache-hit != 'true'
         run: |
-          git clone -c feature.manyFiles=true https://github.com/NOAA-EMC/spack.git
+          git clone -c feature.manyFiles=true https://github.com/JCSDA/spack.git
           source spack/share/spack/setup-env.sh
-          spack env create gsi-monitor-env GSI-Monitor/ci/spack.yaml
+          spack env create gsi-monitor-env gsi-monitor/ci/spack.yaml
           spack env activate gsi-monitor-env
-          spack compiler find
```

Comment:
Is the `spack compiler find` command no longer needed because we are specifying the compiler version?

Reply:
And a related question.  Why not specify the Intel compiler version in spack.yaml like it is for gcc?

Reply:
I am not sure.  I suspect the find is necessary to locate the GCC in case Intel needs it.
I don't know how to specify the compiler version in the spack yaml. May be @AlexanderRichert-NOAA does.

Reply:
Short answer, I would recommend keeping both of those pieces as they are.

`spack compiler find` is needed in order to identify available compilers and store their configurations, the main purpose of which is to let Spack know where to find a given compiler when when it's time to actually build with it. For example, when I tell spack to "install g2c%intel", it will search our inventory of available compilers for a compiler matching "intel" and try to use it, including if I specify an intel version. Some compilers, at least GCC, can be automatically detected by Spack just by running concretize or install or whatever, but I don't think Intel is one of them.

Which gets to the second question-- I seem to recall that the version setting for GCC had something to do with multiple versions of GCC being available, so the idea was to pick one to ensure reproducible/self-consistent runs. By contrast, `spack compiler find` always picks up on only one version of the Intel compilers (I think it's searching $PATH or some other env variable, so it'll only ever find one version), so there's not really a need to specify a version for Intel; either way, Spack will match it in the list of available compilers, and by not specifying it, you don't have to change it each time the version changes in the intel apt repo.

Reply:
That was very informative.  Thanks @AlexanderRichert-NOAA